src/dotnet/ToolPackage/NuGetPackageLocation.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal class NuGetPackageLocation
```

Comment:
Perhaps `NuGetPackageInfo`? This doesn't really encapsulate a particular location.